### DISCUSSION TITLE
Is there equivalent of HTMLLoader Event.HTML_RENDER with StageWebView?

### DISCUSSION THREAD

**Discussion by ltwede** [airsdk/Adobe-Runtime-Support#3300] [Q&A] [answered]
Upgrading an accounting app that runs on Windows from a much earlier version of Air to 50.2. The UseWebView2 tag is set to false. The application doesn't have much web interactivity, but it does interact with a tax authority's website to validate and register a tax ID number. 

The earlier version used the HTMLLoader's HTML_RENDER event, to parse the DOM (document.title property) to retrieve a value sent from the tax authority that indicates that the tax ID is now registered.

I have tried many things with the StageWebView but cannot get this to work. I can see StageWebView has a "title" property, but it doesn't seem to be bindable, so I don't know if/when the value changes. I don't see anything that looks like a render event. The location change event will not work for this because the location doesn't change when the tax authority indicates the tax ID is registered. I tried the 51.0 SDK that has the new config parameter, but I am unsure what properties are available or whether any of them would address this situation. I have tried with UseWebView2 both true and false. I have tried passing true for the useNative parameter and not passing any parameter. I tried listening for DataEvent.WEBVIEW_MESSAGE with the 51 SDK, hoping that might be similar to the HTML_RENDER event. 

I just need something to be aware when the document.title property changes. Any suggestions for what to try?

**Comment by ajwfrost** [comment]
Hi

The basic StageWebView wouldn't give this functionality I'm afraid. Are you able to perhaps use AIR SDK 33.1.1.743, which is the final version we had that includes the webkit library used by HTMLLoader?

Alternatively @marchbold is this something that your webview ANE might be able to handle?

thanks

**Reply by marchbold** [reply, marked as answer]
There is no event dispatched from the NativeWebView extension to indicate the document render is complete however you should be able to run some javascript in the webview using the to monitor the `document.readyState` or I believe you could use an observer on the title directly.

**Reply by ltwede** [reply]
Thank you! 

In my specific case, it turned out I was able to handle by changing my request to the tax authority to use URI redirect instead of OOB_AUTO. This caused the location change event to have the proper information that my app needed.

**Comment by ltwede** [comment]
Thank you. The 33.1.1.743 SDK is a possibility, but what we were trying to address by upgrading was to be up-to-date, with an accompanying warmish-fuzzyish feeling that by keeping up-to-date we would be also keeping the app more secure. 

Very interested in @marchbold 's reply to your question.

Also, any possibility of Harman adding this functionality? It seems like something a host application would be interested in knowing, when the website it was hosting rendered.
